includes/class-wc-google-analytics.php

```diff
@@ -111,6 +113,26 @@ public function __construct() {
 		add_filter( 'woocommerce_get_return_url', array( $this, 'utm_nooverride' ) );
 	}
 
+	/**
+	 * Conditionally display an error notice to the merchant if the stored property ID starts with "UA"
+	 *
+	 * @return void
+	 */
+	public function universal_analytics_upgrade_notice() {
+		if ( 'ua' === substr( strtolower( $this->get_option( 'ga_id' ) ), 0, 2 ) ) {
+			echo sprintf(
+				'<div class="%1$s"><p>%2$s</p></div>',
+				'notice notice-error',
+				sprintf(
+					/* translators: 1) URL for Google documentation on upgrading from UA to GA4 2) URL to WooCommerce Google Analytics settings page */
+					__( 'Your website is configured to use Universal Analytics which Google retired in July of 2023. Update your account using the <a href="%1$s" target="_blank">setup assistant</a> and then update your <a href="%2$s">WooCommerce settings</a>.', 'woocommerce-google-analytics-integration' ), // phpcs:ignore WordPress.Security.EscapeOutput
```

Comment:
I'm thinking if using `esc_html__` would be better even though we are not using any variables in the string. I slightly lean towards to escape it like the code below to show privacy policy so it would be more consistent and also it could keep the string safe if we were to use the variables inside it in the future.

https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/fd97569b48b17a126541bb62096b74a3f61ccafa/includes/class-wc-google-analytics.php#L432-L437

Reply:
Updated in 1f0b6fc